### DISCUSSION TITLE
Expanded POS Functionality

### DISCUSSION THREAD

**Discussion by zdforcier** [btcpayserver/btcpayserver#5515] [Ideas / Feature Requests]
I've been tinkering with BTCPay Server to set up a Bitcoin POS for the business I work at.

Incredible software - very easy to use, and powerful! The Bitcoin side of things works great and there has clearly been a lot of dedication to get it to where it is today.

My one complaint is on the actual POS side of things. The business I work at has a huge menu, and we allow customers to do a lot of modifications and add ons. My ability to create the menu on BTCPay server is pretty limited. 

I’d love it if there was
- options for item notes
- the ability to choose different sized items (small, medium, large)
- the ability to add/remove specific toppings or ingredients 
- the ability to mark items for common allergies (egg, gluten, dairy, etc.)
- the ability to select “extra” toppings, such as double pepperoni or extra cheese

Some of this I’ll try to figure out how to add myself (open source is awesome) but I admit it’s a bit out of my field. If anyone could direct me to some good resources I’d be very appreciative. If anyone has created software which enables these features, that would also be useful.

I’ll start a bounty for these features at 500,000 sats.

Here is the current test site that I've created. I need finish the menu before I can move it over to the main restaurant website though. [https://mimmosbtc.net]

**Comment by dstrukt** [comment]
Did you have an expected, or hopeful timeline in mind?

@dennisreimann, @pavlenex and I have been pushing the front-end of the POS, and I figured we'd eventually add more granular functionality or "modes" like this in the coming months/year, but if you don't expect this immediately, we could tackle a few of these in coming releases.

**Reply by zdforcier** [reply]
No expectations. I have enough to work with that I can make a functional in-store POS. We do a lot of online order volume though, and obviously Bitcoin makes for great "internet money", but I need to do more work to get to the point where I feel comfortable allowing customers to handle placing orders and paying in Bitcoin online. I'm aiming to combine the BTC page with the main website within a year.

The bounty will stand indefinitely. I appreciate the work all you fine folks do and am just excited to start the slow transition away from horrible POS systems like Revel which take a big cut of revenue, have horrible customer support, and only work with fiat.

Funny story - we tried to switch to "Toast" a few months back and had to abandon our plans because of how unusable their backend was and how little customization we could make to the menu. BTCPay is already leagues above Toast in terms of usability and costs.

**Reply by dstrukt** [reply]
Love to hear it, and appreciate the feedback and support, helps us tremendously!

**Comment by caydyan** [comment]
Hi @zdforcier, checking whether the 500,000 sats bounty for expanded POS functionality still stands before investing implementation time.

I am evaluating a scoped BTCPay POS implementation path around item notes plus menu option groups/modifiers for sizes, toppings, extras, and allergy markers. If the bounty is still valid, I can prepare a focused technical plan or start with the smallest mergeable slice maintainers would accept.

No payout request here; I just want to confirm the bounty is still active and whether the original acceptance criteria still match what you need.

**Comment by zdforcier** [comment]
Bounty still stands! I'm ironically no longer currently in the
restaurant industry, but I know this functionality would be useful for all
types of businesses and I would still love to help in funding improvements
to btcpayserver.

- Zach

**Comment by caydyan** [comment]
Submitted the first scoped implementation slice for the active 500,000 sats POS bounty: https://github.com/btcpayserver/btcpayserver/pull/7403

This PR adds item-level notes to POS cart lines, stores them in `posData.cart[].note`, normalizes them server-side, and includes them in receipt cart details. It focuses on the first requested capability from the discussion while keeping pricing and inventory behavior unchanged.

BTC payout address if accepted for the bounty: `bc1qev5ant33v5y89qqjvcf4mh9hlax5svqf5xd7gc`

**Comment by zdforcier** [comment]
Caydyan,
I've got to fire up my old BTCPayserver instance to check on the new
functionality. It's been a while since I've used it and ran it on a hosting
platform, so please allow a few days for me to accept the work and send you
the bounty.
- Zach

**Comment by caydyan** [comment]
Thanks Zach. A few days is fine.

I will keep PR #7403 monitored and handle maintainer review or CI follow-ups while you test. If a specific acceptance checklist or test path would help, I can add it to the PR discussion.

**Comment by caydyan** [comment]
Hi Zach, quick check-in after a few days.

PR #7403 is still open, non-draft, and clean/mergeable at `d1ab3348c81b9da3787d823f6402ff29360f13a5`. The hosted checks were green after the last validation, and I added a manual POS acceptance checklist here:

https://github.com/btcpayserver/btcpayserver/pull/7403#issuecomment-1730113894

If the item-notes slice works for your BTCPay POS test and you consider it acceptable for the 500,000 sats bounty, BTC payout address is:

`bc1qev5ant33v5y89qqjvcf4mh9hlax5svqf5xd7gc`

No rush if you are still setting up the instance; I am keeping the PR monitored for maintainer review or test feedback.

**Comment by caydyan** [comment]
Hi Zach, quick status sync after the weekend.

PR #7403 is still open, non-draft, clean/mergeable at `d1ab3348c81b9da3787d823f6402ff29360f13a5`, with CodeRabbit and the CircleCI `build_and_test` aggregate still passing. The manual POS acceptance checklist is on the PR here:

https://github.com/btcpayserver/btcpayserver/pull/7403#issuecomment-1730113894

If you have had a chance to test and this item-notes slice is acceptable for the 500,000 sats POS bounty, the BTC payout address remains:

`bc1qev5ant33v5y89qqjvcf4mh9hlax5svqf5xd7gc`

If you are still setting up the old instance, no problem; I am keeping the PR monitored for maintainer or sponsor feedback.

**Comment by Latrahisondesimages** [comment]
Before starting another implementation: is the original 500,000 sats bounty already fully allocated to the item-notes PR #7403, or is there still a separate bounty for one of the remaining requested capabilities (sizes, toppings/extras, or allergy markers)? I can take a small mergeable slice, but I want to confirm the exact paid acceptance scope first and avoid duplicating the existing work.
